vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/RenderingManager.java

```diff
@@ -0,0 +1,315 @@
+
+        } catch (SlidesInSequenceNotFoundException  | SequenceNotFoundException |SlideNotFoundException e ) {
+            logger.error("Could not add html page for slide" , e);
+        }       
+    }
+
+    
+    /**
+     *@see IRenderingManager#populateContextForSlide(Context, String, String, String, String)
+     */
+    @Override
+    public void populateContextForSlide(Context context, String spaceId, String moduleId, String sequenceId, String slideId) 
+            throws SlidesInSequenceNotFoundException, SequenceNotFoundException, SlideNotFoundException {
+        IModule module = moduleManager.getModule(moduleId);
+        context.setVariable("module", module);
+        String startSequenceId = module.getStartSequence().getId();
+
+        context.setVariable("startSequenceId", startSequenceId);
+        ISequence sequenceExist=moduleManager.checkIfSequenceExists(moduleId, sequenceId);
+        if (sequenceExist==null) {
+            throw new SequenceNotFoundException(sequenceId);
+        }
+        List<ISlide> sequenceSlides = sequenceManager.getSequence(sequenceId).getSlides();
+
+        boolean slideExist = sequenceSlides.stream().anyMatch(slide -> slide.getId().equals(slideId));
+        if (!slideExist) {
+            throw new SlideNotFoundException(slideId);
+        }
+
+        if (sequenceSlides.size() == 0) {
+            throw new SlidesInSequenceNotFoundException();
+        }
+        context.setVariable("firstSlide", module.getStartSequence().getSlides().get(0).getId());
+        
+        ISlide currentSlide = slideManager.getSlide(slideId);
+        int slideIndex = sequenceSlides.indexOf(currentSlide);
+
+        int slideSize = sequenceSlides.size();
+        
+        String prevSlideId = (slideIndex > 0) ? sequenceSlides.get(slideIndex - 1).getId() : "";
+        String nextSlideId = (slideSize > slideIndex + 1) ? sequenceSlides.get(slideIndex + 1).getId() : "";
+
+        context.setVariable("sequences",moduleManager.getModuleSequences(moduleId));
+        context.setVariable("sequence",sequenceExist);
+        context.setVariable("slides", sequenceSlides);
+        context.setVariable("currentSequenceId", sequenceId);
+        context.setVariable("nextSlide", nextSlideId);
+        context.setVariable("prevSlide", prevSlideId);
+
+        context.setVariable("currentSlideCon", currentSlide);
+
+        context.setVariable("numOfSlides", sequenceSlides.size());
+        context.setVariable("currentNumOfSlide", slideIndex + 1);
+        context.setVariable("spaceId", spaceId);
+        context.setVariable("spaceName", spaceManager.getSpace(spaceId).getName());
+
+    }
+    
+    /** Populates context with variables to process space template
+     * 
+     * @param context
+     * @param id
+     */
+    @Override
+    public void populateContextForSpace(Context context, String id, SequenceHistory sequenceHistory) {
```

Comment:
why public?

Reply:
this is being used in unit tests

Reply:
It should not be public just for the unit tests.